apps/typeform/frontend/src/index.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { render } from 'react-dom';
+import {
+  init,
+  locations,
+  AppExtensionSDK,
+  FieldExtensionSDK,
+  DialogExtensionSDK
+} from 'contentful-ui-extensions-sdk';
+import '@contentful/forma-36-react-components/dist/styles.css';
+import '@contentful/forma-36-fcss/dist/styles.css';
+import Authentication from './Auth';
+import { TypeFormField } from './Field/TypeFormField';
+import { TypeformPreviewWidget } from './TypeFormWidget';
+import './index.scss';
+import processTokenCallback from './processTokenCallback';
+
+if (window.location.search.includes('token')) {
+  processTokenCallback(window);
+} else {
+  init(sdk => {
+    if (sdk.location.is(locations.LOCATION_APP_CONFIG)) {
+      render(<Authentication sdk={sdk as AppExtensionSDK} />, document.getElementById('root'));
+    } else if (sdk.location.is(locations.LOCATION_DIALOG)) {
+      render(
+        <TypeformPreviewWidget sdk={sdk as DialogExtensionSDK} />,
+        document.getElementById('root')
+      );
+    } else {
+      render(
+        <TypeFormField sdk={sdk as AppExtensionSDK & FieldExtensionSDK} />,
```

Comment:
`AppExtensionSDK` is available only in `LOCATION_APP_CONFIG` and you handle it earlier: https://github.com/contentful/apps/pull/65/files#diff-8bd9bee2ac78785f05f6a7d94ee9f546R22-R23